vllm/model_executor/layers/layernorm.py

```diff
@@ -162,7 +196,7 @@ def forward_cuda(
             return self.forward_native(x, residual)
 
         add_residual = residual is not None
-        norm_func = dispatch_cuda_rmsnorm_func(add_residual)
+        norm_func = dispatch_rmsnorm_func(add_residual, self.dtype)
```

Comment:
Instead of dispatching inside `forward_cuda`, please add `forward_rocm`. Also the dispatching can happen inside __init__ instead of during runtime, that's the benefit of the `CustomOp` abstraction.

Reply:
> it would be redundant to implement the forward_hip with the same code

I think here separation of concerns is more important than DRY (don't repeat yourself).

> `forward_hip` defaults to the CUDA implementation and it is documented in the class

Yes, but here we're changing the default behavior, so it's okay to override the `forward_hip` method.